core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/AbstractCamelContext.java

```diff
@@ -3786,7 +3787,9 @@ public DataFormat resolveDataFormat(String name) {
         final DataFormat answer = dataformats.computeIfAbsent(name, new Function<String, DataFormat>() {
             @Override
             public DataFormat apply(String s) {
-                DataFormat df = ResolverHelper.lookupDataFormatInRegistryWithFallback(getCamelContextReference(), name);
+                DataFormat df = Optional
+                        .ofNullable(ResolverHelper.lookupDataFormatInRegistryWithFallback(getCamelContextReference(), name))
+                        .orElse(getDataFormatResolver().createDataFormat(name, getCamelContextReference()));
```

Comment:
You should eiter usean old school null check or use `orElseGet` instead of `orElse` as the later will cause the creation of a data format even if the former may return something

Reply:
We can also remove the ternary operation on the `return`